### PR TITLE
perf(tracing/request-id): introduce $kong_request_id

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

The version of lua-kong-module has bumped to 0.8.0, introducing a new variable `$kong_request_id`.

KAG-2734

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* [Implement ...]

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

**Comment by zhongweiy** [comment]
It will be helpful to have performance data after applying this patch and https://github.com/Kong/lua-kong-nginx-module/pull/70/files. 

Here is some performance data done before for reference.
1. https://konghq.atlassian.net/browse/KAG-2692?focusedCommentId=113705
2. nginx_config is in the description of this ticket https://konghq.atlassian.net/browse/KAG-2692 and wrk command: `wrk -t1 -c40 -d10s http://127.0.0.1:80`

**Review by chobits — APPROVED** [review]

**Review by samugi — APPROVED** [review]

File: kong/tracing/request_id.lua [hunk 1]
```diff
@@ -21,14 +23,14 @@ local function get()
...
     if not NGX_VAR_PHASES[phase] then
       return nil, "cannot access ngx.var in " .. phase .. " phase"
     end
 
     -- first access to the request id for this request:
-    -- initialize with the value of $request_id
-    rid = ngx.var.request_id
+    -- initialize with the value of $kong_request_id
+    rid = var.kong_request_id
     ngx.ctx.request_id = rid
```
[the remark this answers is not in the document]
Comment (chronolaw) [reply kong/tracing/request_id.lua:34]: Yes, I think it is still needed, accessing nginx variable is a relatively expensive operation, even if it is indexed.

**Review by zhongweiy — APPROVED** [review]

**Comment by zhongweiy** [comment]
Shall we also update the FIPS document like "core/uuid" does? https://docs.konghq.com/gateway/latest/kong-enterprise/fips-support/

**Comment by fffonion** [comment]
The FIPS document is out-of-date, I will create a PR that updates it along with the request_id change.

**Comment by ADD-SP** [comment]
Please rebase on the master and adapt to changes of https://github.com/Kong/kong/pull/11017, that PR also uses `$request_id`. @chronolaw

**Review by ADD-SP — CHANGES_REQUESTED** [review]
Please rebase on the master and adapt to changes of https://github.com/Kong/kong/pull/11017, that PR also uses $request_id.

cc @samugi

**Comment by chronolaw** [comment]
> Please rebase on the master and adapt to changes of #11017, that PR also uses $request_id.
> 
> cc @samugi

Thank you @ADD-SP , I have rebased the branch and fixed some code in #11017.

**Review by samugi — APPROVED** [review]

**Comment by chronolaw** [comment]
I also found another usage or `var.request_id` in `plugins/aws-lambda/request-util.lua`, should we change it too?

File: kong/tools/request_aware_table.lua [hunk 2]
```diff
@@ -33,7 +33,7 @@ local function enforce_sequential_access(table)
     return
   end
 
-  local curr_request_id = var.request_id
+  local curr_request_id = var.kong_request_id
```
Comment (chronolaw) [inline kong/tools/request_aware_table.lua:36]: Perhaps `tracing.request_id.get()` is a better choice? @samugi.

Comment (ADD-SP) [reply kong/tools/request_aware_table.lua:36]: +1 for `tracing.request_id.get()`, but don't forget to cache this function.

Comment (chronolaw) [reply kong/tools/request_aware_table.lua]: done

Comment (samugi) [reply kong/tools/request_aware_table.lua]: yes that's probably better @chronolaw

**Comment by bungle** [comment]
@chronolaw I think we should change this too:
https://github.com/Kong/kong/blob/master/kong/plugins/aws-lambda/request-util.lua#L224

File: changelog/unreleased/kong/kong_request_id.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+message: Bumped lua-kong-nginx-module from 0.7.1 to 0.8.0
```
Comment (dndx) [inline changelog/unreleased/kong/kong_request_id.yml:1]: The filename of this file should be changed. Maybe `bump_lua_kong_nginx_module.yml`

**Review by dndx — CHANGES_REQUESTED** [review]

File: changelog/unreleased/kong/kong_request_id.yml (hunk 3 above)
Comment (chronolaw) [reply changelog/unreleased/kong/kong_request_id.yml]: done.

**Comment by chronolaw** [comment]
> @chronolaw
> 
> ```
>    ERROR  spec/03-plugins/27-aws-lambda/05-aws-serializer_spec.lua:46: [AWS Lambda] aws-gateway input serializes a request regex
> ./kong/tracing/request_id.lua:26: attempt to call upvalue 'get_phase' (a nil value)
> 
> stack traceback:
> 	./kong/tracing/request_id.lua:26: in function 'get_request_id'
> 	./kong/plugins/aws-lambda/request-util.lua:225: in function 'aws_serialize'
> 	spec/03-plugins/27-aws-lambda/05-aws-serializer_spec.lua:80: in function <spec/03-plugins/27-aws-lambda/05-aws-serializer_spec.lua:46>
> ```

The reason is that now we are using `tracing.request_id`, which needs to call `ngx.get_phase()`.
I will add mock function in test case.

File: changelog/unreleased/kong/kong_request_id.yml (hunk 3 above)
Comment (chronolaw) [reply changelog/unreleased/kong/kong_request_id.yml]: We already have a `lua_kong_nginx_module_bump.yml` in PR #11663, should we rename it to another name?

Comment (ADD-SP) [reply changelog/unreleased/kong/kong_request_id.yml]: How about `bump_lua_kong_nginx_module_from_0_7_1_to_0_8_0.yml`?

Comment (chronolaw) [reply changelog/unreleased/kong/kong_request_id.yml]: reuse that bump yml file.

**Comment by dndx** [comment]
@chronolaw Why do we `skip-changelog` this PR?

**Review by dndx — APPROVED** [review]

**Review by ADD-SP — APPROVED** [review]

**Pull request merged**